js/ui/appIconBar.js

```diff
@@ -641,10 +641,22 @@ const ScrolledIconList = GObject.registerClass({
 
     setActiveApp(app) {
         this._taskbarApps.forEach((appButton, taskbarApp) => {
-            if (app == taskbarApp)
-                appButton.add_style_pseudo_class('highlighted');
-            else
-                appButton.remove_style_pseudo_class('highlighted');
+            var changed = false;
```

Comment:
Does this change fix https://phabricator.endlessm.com/T26981?

Reply:
Yes, looks like it from a quick test. The initial problem I saw was that closing the app would not update the indicator but with this change it also seems to fix the other issues pointed in the task above.

Reply:
Yes, confirmed it fixes this issue - updated the task accordingly.

Reply:
Please update the commit message and PR summary with the URL to that ticket in that case!

Reply:
Done.